services/src/kilda-pce/src/main/java/org/openkilda/pce/impl/PathComputerImpl.java

```diff
@@ -0,0 +1,169 @@
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.pce.impl;
+
+import static java.lang.String.format;
+
+import org.openkilda.model.Flow;
+import org.openkilda.model.Isl;
+import org.openkilda.model.Node;
+import org.openkilda.model.Path;
+import org.openkilda.pce.PathComputer;
+import org.openkilda.pce.PathPair;
+import org.openkilda.pce.RecoverableException;
+import org.openkilda.pce.UnroutableFlowException;
+import org.openkilda.pce.impl.algo.SimpleGetShortestPath;
+import org.openkilda.pce.impl.model.AvailableNetwork;
+import org.openkilda.pce.impl.model.SimpleIsl;
+import org.openkilda.persistence.PersistenceException;
+import org.openkilda.persistence.repositories.IslRepository;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.LinkedList;
+import java.util.List;
+
+@Slf4j
+public class PathComputerImpl implements PathComputer {
+    private final IslRepository islRepository;
+
+    public PathComputerImpl(IslRepository islRepository) {
+        this.islRepository = islRepository;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PathPair getPath(Flow flow, Strategy strategy, boolean allowSameFlowPath)
+            throws UnroutableFlowException, RecoverableException {
+        return getPath(flow, strategy, allowSameFlowPath, 35);
+    }
+
+    /**
+     * Gets path between source and destination switch for specified flow.
+     *
+     * @param flow              the {@link Flow} instance.
+     * @param strategy          the path find strategy.
+     * @param allowSameFlowPath whether to allow the existing flow path to be a potential new path or not.
+     * @param allowedDepth      the allowed depth for a potential path.
+     * @return {@link PathPair} instances
+     */
+    public PathPair getPath(Flow flow, Strategy strategy, boolean allowSameFlowPath, int allowedDepth)
+            throws UnroutableFlowException, RecoverableException {
+
+        AvailableNetwork network = new AvailableNetwork();
+        try {
+            // Reads all active links from the database and creates representation of the network
+            // in the form of {@link SimpleSwitch} and {@link SimpleIsl} between them.
+            Iterable<Isl> links = flow.isIgnoreBandwidth() ? islRepository.findActive() :
+                    islRepository.findActiveWithAvailableBandwidth(flow.getBandwidth());
+            links.forEach(isl -> network.addLink(isl.getSrcSwitchId(), isl.getDestSwitchId(),
+                    isl.getSrcPort(), isl.getDestPort(),
+                    isl.getCost(), isl.getLatency()));
+
+            if (allowSameFlowPath) {
```

Comment:
The reason for these changes is to hide implementation details behind PathComputer interface. AvailableNetwork must be removed from the interface definition as only related to in-memory Dijkstra's algorithm, it may not be applicable to other implementations.

As for now there are 2 criteria: flow and allowSameFlowPath. And if we're going to add more in the future then a separate structure can be introduced which absorbs all criteria. However this is out of scope of this PR.